nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-nar-utils/src/main/java/org/apache/nifi/nar/StandardExtensionDiscoveringManager.java

```diff
@@ -169,6 +181,75 @@ public void discoverExtensions(final Set<Bundle> narBundles, final boolean logDe
         }
     }
 
+    public void setPythonBridge(final PythonBridge pythonBridge) {
+        this.pythonBridge = pythonBridge;
+    }
+
+    @Override
+    public void discoverPythonExtensions(final Bundle pythonBundle) {
+        logger.info("Scanning to discover which Python extensions are available and importing any necessary dependencies. This may take a few minutes. See python logs for more details.");
```

Comment:
👍 Will add a qualifier to indicate that it may take a few minutes if there are new extensions discovered. I would avoid mentioning `python.log` explicitly because it's configurable.